packages/neuron-wallet/src/services/transaction-sender.ts

```diff
@@ -91,14 +93,17 @@ export default class TransactionSender {
     return this.broadcastTx(walletID, tx)
   }
 
-  public async broadcastTx(walletID: string = '', tx: Transaction) {
+  public async broadcastTx(walletID: string = '', tx: Transaction, amendHash = '') {
     const currentNetwork = NetworksService.getInstance().getCurrent()
     const rpc = generateRPC(currentNetwork.remote, currentNetwork.type)
     await rpc.sendTransaction(tx.toSDKRawTransaction(), 'passthrough')
     const txHash = tx.hash!
 
     await TransactionPersistor.saveSentTx(tx, txHash)
     await MultisigService.saveSentMultisigOutput(tx)
+    if (amendHash) {
+      await AmendTransactionService.save(txHash, amendHash)
+    }
```

Comment:
Why not move these codes to https://github.com/nervosnetwork/neuron/blob/fix-332/packages/neuron-wallet/src/controllers/wallets.ts#L446? After `sendTx` success, save `AmendTransaction` entity if this is an amend transaction,

Reply:
I guess adding the `amendHash` here is used to support others' transaction types. So I close this conversation.